src/js/components/Button/StyledButtonKind.js

```diff
@@ -0,0 +1,221 @@
+import styled, { css } from 'styled-components';
+
+import {
+  backgroundStyle,
+  focusStyle,
+  genericStyles,
+  normalizeColor,
+} from '../../utils';
+import { defaultProps } from '../../default-props';
+
+const radiusStyle = props => {
+  const size = props.sizeProp;
+  if (size && props.theme.button.size && props.theme.button.size[size]) {
+    return props.theme.button.size[size].border.radius;
+  }
+  return props.theme.button.border.radius;
+};
+
+const fontStyle = props => {
+  const size = props.sizeProp || 'medium';
+  const data = props.theme.text[size];
+  return css`
+    font-size: ${data.size};
+    line-height: ${data.height};
+  `;
+};
```

Comment:
My only concern is that the function is duplicated on StyledButton.js, and any future enhancements for this function would need to be applied on both hence might be missed for either one.

Reply:
I thought about this. But, you'll notice that `plainStyle()` is slightly different between the two. I was more concerned that unintended side-effects would happen changing a single copy. I was trying to maintain separation between old and new approaches.